consul/consul.go

```diff
@@ -169,25 +183,53 @@ func (c *Consul) register(service *consulapi.AgentServiceRegistration) error {
 }
 
 func (c *Consul) DeregisterByTask(taskId apps.TaskId, agentAddress string) error {
-	service, err := c.findServiceByTaskId(taskId)
+	services, err := c.findServicesByTaskId(taskId)
 	if err != nil {
 		return err
+	} else if len(services) == 0 {
+		return errors.New(fmt.Sprintf("Couldn't find any service matching task id %s", taskId))
 	}
-	return c.Deregister(service)
+	return c.deregisterMultipleServices(services, taskId)
 }
 
-func (c *Consul) findServiceByTaskId(searchedTaskId apps.TaskId) (*service.Service, error) {
-	services, err := c.GetAllServices()
-	if err != nil {
-		return nil, err
-	}
+func (c *Consul) deregisterMultipleServices(services []*service.Service, taskId apps.TaskId) error {
+	var deregisterErrors []error
 	for _, s := range services {
-		taskId, err := s.TaskId()
-		if err == nil && taskId == searchedTaskId {
-			return s, nil
+		deregisterErr := c.Deregister(s)
+		if deregisterErr != nil {
```

Comment:
Yeah ;/ I had no idea how to test that in the current testing infrastructure. Any ideas?